packages/shared/src/components/cards/ActionsButtons/UpvoteButtonIcon.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { withExperiment } from '../../withExperiment';
+import { feature } from '../../../lib/featureManagement';
+import { UpvoteIcon } from '../../icons';
+import { IconProps, IconSize } from '../../Icon';
+import { userPrefersReducedMotions } from '../../../styles/media';
+import { useMedia } from '../../../hooks';
+import styles from './ActionButtons.module.css';
+
+type AnimatedButtonIconProps = IconProps & { userClicked?: boolean };
+
+const arrows = Array.from({ length: 5 }, (_, i) => i + 1);
+
+const AnimatedButtonIcon = (props: AnimatedButtonIconProps): ReactElement => {
+  const { secondary: isUpvoteActive, userClicked } = props;
+
+  return (
+    <span className="pointer-events-none relative">
+      <UpvoteIcon secondary={isUpvoteActive} />
+      {userClicked && isUpvoteActive && (
+        <span
+          aria-hidden
+          className={classNames(
+            styles.upvotes,
+            'absolute left-1/2 top-0 h-full w-[125%] -translate-x-1/2',
+          )}
+          role="presentation"
+        >
+          {arrows.map((i) => (
+            <UpvoteIcon
+              secondary
+              size={IconSize.XXSmall}
+              className={styles.upvote}
+              key={i}
+            />
+          ))}
+        </span>
+      )}
+    </span>
+  );
+};
+
+const AnimatedUpvoteButtonIcon = React.memo(
+  function AnimatedUpvoteButtonIconComp(
+    props: AnimatedButtonIconProps,
+  ): ReactElement {
+    const haveUserPrefersReducedMotions = useMedia(
+      [userPrefersReducedMotions.replace('@media', '')],
+      [false],
+      false,
+      false,
+    );
+
+    const Experiment = withExperiment(AnimatedButtonIcon, {
```

Comment:
This will create a new component each time which will cause re-mount on each parent re-render, maybe better to `useConditonalFeature` here?

Are you using `memo` because of that?

Reply:
He wants either the regular component or this animated one.
I suggested your `withExperiment` was probably best one right?

(With addition it needs to be conditional (so he added it inside))
Does that work?

Reply:
Yeah, but `withExperiment` should not be defined inline, it should be const outside.

Reply:
I would say in this case it's better to use `useConditionalFeature` since he needs other dynamic data for user prefers motion.

Reply:
Oh got it, I guess he did it because he needed the conditional eveluation 😅
Guess that brings trouble then?

So we'd need to maybe move back (Sorry Luca 😅)

Reply:
⬆️  😅

Reply:
Okay, let's revert 😢

Reply:
@rebelchris @nensidosari @capJavert updated PR!